### PR TITLE
fix: avoid escaping bind-like params containing colons

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY

A recent PR #17111 that fixed 500s being returned when a virtual table query contained a word prefixed by a colon (e.g. ":foo") caused a regression that unnecessarily also escaped similar strings that contained/ended in a colon (e.g. ":foo:" or ":foo:bar"). 

This PR simplifies the escaping logic by simply escaping all colons with a backslash when `sa.text` is used, as these elements are never expected to contain binds. In addition, an integration test is added that contains colons in the virtual table query, metric, filter and where clauses. The new test fails on current master with a 500.

### AFTER
Now all strings including colons are displayed properly on the query:
image

### BEFORE
Before the virtual table was incorrectly escaped in multiple places:
image

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/connectors/sqla/models.py [hunk 1]
```diff
@@ -930,9 +938,8 @@ def _get_sqla_row_level_filters(
         filters_grouped: Dict[Union[int, str], List[str]] = defaultdict(list)
         try:
             for filter_ in security_manager.get_rls_filters(self):
-                clause = text(
-                    f"({template_processor.process_template(filter_.clause)})"
-                )
+                rendered_filter = template_processor.process_template(filter_.clause)
+                clause = text(f"({rendered_filter})")
```
Comment (john-bodley) [inline superset/connectors/sqla/models.py:942]:
```suggestion
                clause = text(f"({template_processor.process_template(filter_.clause)})")
```

Maybe just inline this given that `rendered_filter` isn't used elsewhere.

File: tests/integration_tests/charts/api_tests.py [hunk 2]
```diff
@@ -2033,3 +2035,58 @@ def test_chart_data_series_limit(self):
...
+            }
+        ]
+        request_payload["queries"][0]["filters"] = [
+            {"col": "foo", "op": "!=", "val": ":qwerty:",}
+        ]
+
+        rv = self.post_assert_metric(CHART_DATA_URI, request_payload, "data")
+        db.session.delete(table)
+        db.session.commit()
+        self.assertEqual(rv.status_code, 200)
```
Comment (john-bodley) [inline tests/integration_tests/charts/api_tests.py:2082]:
```suggestion
        assert rv.status_code == 200
```

**Review by john-bodley — APPROVED** [review]
LGTM, with one non-blocking comment. Its somewhat strange that we need to handle this logic, i.e., blindly I would have assumed that SQLAlchemy would deal with the escaping.

**Comment by villebro** [comment]
Thanks for the review @john-bodley, good comments to my oversights!

> LGTM, with one non-blocking comment. Its somewhat strange that we need to handle this logic, i.e., blindly I would have assumed that SQLAlchemy would deal with the escaping.

AFAIK, currently there is no way of indicating that the clause is literal (like `literal_column`, for instance). Hence colons need to be escaped to indicate to SQLAlchemy that they should not be made available as bind parameters during compilation. But I'm going to propose adding `is_literal` to the `text()` factory, and then adding a `literal_text` factory as shorthand to match (`literal_column`), as I've seen multiple other people ask for the same thing elsewhere (in all the posts I've found, the author of SQLAlchemy always recommends just escaping the bind-like parameters).

**Pull request merged**